### PR TITLE
Eason/129 add timeline to resource

### PR DESCRIPTION
**Pull request by WYC401**
## Description

- add timeline page into IDEO resource page
- use the value read from .env file in timeline to be free from specific server
-  change the readme to guide user on connecting to ec2 server or local server

## Updates

Under Review.

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

### CODE REVIEW

**Comment by WYC401** [comment]
[add timeline into private resource tab in ideo project and auto sort with ideo project tab](https://trello.com/c/AT7n3DXi/129-add-timeline-into-private-resource-tab-in-ideo-project-and-auto-sort-with-ideo-project-tab)

File: README.md [hunk 1]
```diff
@@ -4,3 +4,10 @@ Documentation web-app built by the coding team. Features a timeline to document
 # Command to run frontend
 npm install
 npm start
+
+# Connect to backend
+- if connected to the backend running on the local machine, add a `.env` file in the root folder and copy the following line:
+```
+REACT_APP_API_URL=http://localhost:4000
+```
```
Comment (svolterra) [inline README.md:12]: Nice!

File: src/pages/Project/ProjectResources.tsx [hunk 2]
```diff
@@ -3,18 +3,38 @@ import { Project } from '@entities/Project';
...
 
 const ProjectDefault: React.FC<ProjectProps> = (props) => {
-    return (
-        <div className='project-subcontent-container'>
-            <ProjectBreadcrumbs project_name={props.project.name} page_name={TEXT.PROJECT_NAV.RESOURCES} /> 
-            resources page for {props.project.name};
-        </div>
-    )
+    
+    const timeLineDefaultFilter: SearchFilter = {
```
Comment (svolterra) [inline src/pages/Project/ProjectResources.tsx:14]: Just one comment here is that the naming is a little misleading as it's the same const name used in the `DefaultTimeline.tsx` file. How about something like `resourcesTimelineFilter` or `projectTimelineFilter`? :)

File: src/pages/Project/ProjectResources.tsx [hunk 3]
```diff
@@ -3,18 +3,38 @@ import { Project } from '@entities/Project';
...
+            <div className='project-subcontent-container'>
+                <ProjectBreadcrumbs project_name={props.project.name} page_name={TEXT.PROJECT_NAV.RESOURCES} /> 
+                resources page for {props.project.name};
+            </div>
+        )
+    } else {
+        return (
+            <div className='project-subcontent-container'>
+                <ProjectBreadcrumbs project_name={props.project.name} page_name={TEXT.PROJECT_NAV.RESOURCES} /> 
+                <Timeline defaultFilter={timeLineDefaultFilter}/>
```
Comment (svolterra) [inline src/pages/Project/ProjectResources.tsx:33]: Similar thing here: how about changing the prop field to `projectFilter` instead? :) The name `defaultFilter` did confuse me a little.

**Review by svolterra — COMMENTED** [review]
Nice work Eason! The IDEO resources page looks great :) My one comment here is that I was having a hard time following the differences between `Timeline` and `DefaultTimeline` because to me the names of the props and such were so similar. So I left a couple of suggestions in my code review for your consideration 😄 Otherwise, looks good!

File: src/pages/Project/ProjectResources.tsx (hunk 2 above)
Comment (WYC401) [reply src/pages/Project/ProjectResources.tsx]: has changed to `resourcesTimelineFilter `

File: src/pages/Project/ProjectResources.tsx (hunk 3 above)
Comment (WYC401) [reply src/pages/Project/ProjectResources.tsx]: has changed to `resourcesTimelineFilter `

File: package.json
Comment (ElisaHu1) [inline package.json:1]: change this back plz

File: src/pages/Timeline/DefaultTimeline.tsx
Comment (ElisaHu1) [inline src/pages/Timeline/DefaultTimeline.tsx:1]: why are you hard coding a default timeline?

File: src/pages/Timeline/Timeline.tsx [hunk 4]
```diff
@@ -63,7 +69,7 @@ const Timeline: React.FC<TimelineProps> = (props) => {
   }
 
   const deleteCommit = async (_id: string) => {   
-    return axios.delete(`http://localhost:4000/api/snapshots/${_id}`,  { 
+    return axios.delete(`${baseURL}/api/snapshots/${_id}`,  {
```
Comment (ElisaHu1) [inline src/pages/Timeline/Timeline.tsx:72]: nice change

File: package-lock.json
Comment (ElisaHu1) [inline package-lock.json:1]: why are you adding peer = true?

**Comment by ElisaHu1** [comment]
overall looks good, but the timeline oversize the regular layout, are you able to zoom out the timeline just for the project? 
image
also can you fix the conflict?

File: src/pages/Timeline/DefaultTimeline.tsx
Comment (WYC401) [reply src/pages/Timeline/DefaultTimeline.tsx:1]: In my refactoring, the previous `Timeline.tsx `became a base/abstract react class, which required the` defaultfilter` in props to be set. With this flexibility, we can set default filter when people enter the timeline page. So, I can make adjustment on `<Timeline>`'s props to make both Timeline Page and Timeline Page in resource tab of IDEO project. Current `DefaultTimeline.tsx` is actually the previous` Timeline.tsx.`

File: package-lock.json
Comment (WYC401) [reply package-lock.json:1]: Right. I have reverted it back by copying and pasting the file in main branch

**Comment by WYC401** [comment]
> overall looks good, but the timeline oversize the regular layout, are you able to zoom out the timeline just for the project? image also can you fix the conflict?

I have adjusted it to 80% of its original size. Also, I have resolved the conflict by rebasing the main branch

**Review by ElisaHu1 — APPROVED** [review]
looks good to me

**Comment by ElisaHu1** [comment]
your rebase is not done properly so the github showing 35 files changed, it will be very hard for us to review your real change. Are you able to fix that? @WYC401

**Pull request closed without merging**